libraries/GCS_MAVLink/GCS.h

```diff
@@ -470,11 +465,80 @@ class GCS_MAVLINK
     char _perf_packet_name[16];
     char _perf_update_name[16];
 
-    // deferred message handling.  We size the deferred_message
-    // ringbuffer so we can defer every message type
-    enum ap_message deferred_messages[MSG_LAST];
-    uint8_t next_deferred_message;
-    uint8_t num_deferred_messages;
+    // outbound ("deferred message") queue.
+
+    // "special" messages such as heartbeat, next_param etc are
+    // stored separately to stream-rated messages like AHRS2 etc:
+    struct deferred_message_t {
+        const ap_message id;
+        uint16_t interval_ms;
+        uint16_t last_sent_ms; // from AP_HAL::millis16()
+    } deferred_message[2] = {
+        { MSG_HEARTBEAT, },
+        { MSG_NEXT_PARAM, },
+    };
+    // returns index of id in deferred_message[] or -1 if not present
+    int8_t deferred_message_offset(const ap_message id) const;
```

Comment:
a small thing but I might have named it "_index" instead of "_offset".  for me offsets are for sensors, indexes are for the position in an array.

Reply:
Changed.